File: backend/app/schemas/keywords.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class SearchKeywordBase(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    keyword: str
    category: str | None = None
    brand: str | None = None
    model_group: str | None = None
    priority: int = 100
    active: bool = True


class SearchKeywordCreate(SearchKeywordBase):
    pass


class SearchKeywordUpdate(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    keyword: str | None = None
    category: str | None = None
    brand: str | None = None
    model_group: str | None = None
    priority: int | None = None
    active: bool | None = None


class SearchKeywordRead(SearchKeywordBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, protected_namespaces=())

File: frontend/src/types/keywords.ts
export type SearchKeyword = {
  id: number
  keyword: string
  category: string | null
  brand: string | null
  model_group: string | null
  priority: number
  active: boolean
  created_at: string
  updated_at: string
}

export type SearchKeywordCreate = {
  keyword: string
  category?: string | null
  brand?: string | null
  model_group?: string | null
  priority?: number
  active?: boolean
}
